saleor/graphql/warehouse/mutations.py

```diff
@@ -0,0 +1,84 @@
+import graphene
+
+from saleor.account.models import Address
+from saleor.graphql.core.mutations import ModelDeleteMutation, ModelMutation
+from saleor.graphql.core.types.common import WarehouseError
+from saleor.graphql.warehouse.types import WarehouseCreateInput, WarehouseUpdateInput
+from saleor.warehouse import models
+
+ADDRESS_FIELDS = [
+    "street_address_1",
+    "street_address_2",
+    "city",
+    "city_area",
+    "postal_code",
+    "country",
+    "country_area",
+    "phone",
+]
+
+
+class WarehouseCreate(ModelMutation):
+    class Arguments:
+        input = WarehouseCreateInput(
+            required=True, description="Fields required to create warehouse."
+        )
+
+    class Meta:
+        description = "Creates new warehouse."
+        model = models.Warehouse
+        permissions = ("warehouse.manage_warehouses",)
+
+    @classmethod
+    def create_address(cls, clean_data):
+        address_data = clean_data["address"]
+        address = Address.objects.create(**address_data)
+        return address
+
+    @classmethod
+    def clean_input(cls, info, instance, data, input_cls=None):
+        cleaned_data = super().clean_input(info, instance, data, input_cls=input_cls)
+        cleaned_data["address"] = cls.create_address(cleaned_data)
```

Comment:
I am not sure that we should create an instance of an address inside a cleaning method. 
Maybe `construct_instance`,  `save` or `_save_m2m` would be a better place for that?